### PR TITLE
Feature 2368 helm k8s

### PR DESCRIPTION
**Pull request by artntek**
Metacatui Helm Chart:
* Change Matt's prototype to check out static content from GitHub on deploy
* Add configmap for metacatui's config/config.js, with values overridable in values.yaml

### CODE REVIEW

File: helm/values.yaml [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  ## Optional configuration -- leave commented to use metacatui default values
+  ##
+#  d1CNBaseUrl: "https://cn-sandbox.test.dataone.org/cn"
+#  mapKey: "your-map-key-here"
+#  mdqBaseUrl: "https://localhost:8080:30443/quality"
+#  dataoneSearchUrl: "https://search-stage.test.dataone.org"
+#  portalLimit: 100
+#  googleAnalyticsKey: "your-google-analytics-key-here"
+#  bioportalAPIKey: "your-bio-portal-api-key-here"
+#  cesiumToken: "your-cesium-token-here"
```
Comment (robyngit) [inline helm/values.yaml:39]: Can we set any of the appConfig values here?

**Comment by robyngit** [comment]
Thanks for working on this @artntek !! Besides the in-line comments above, here are some general questions and thoughts that I came up looking through this:

- There are two setups we generally use for deployments:
   1. `index.html` points directly to a config that exists in the themes directory, `e.g. src/js/themes/{THEME-NAME}/config.js`
   2. `index.html` points to config/config.js - this config sets the theme and basically extends the `src/js/themes/{THEME-NAME}/config.js` file.
  
  In the first case, we [update the index.html file](https://github.nceas.ucsb.edu/NCEAS/metacatui-config/blob/e1a57509a3b72bcf6a35568fe2a6e36c19ddf403/scripts/upgrade.sh#L200-L203) to point to the theme config. I'm wondering if we can still do either type of deployment with the helm chart? Or would editing index.html be a manual step for now?

- We should also add a guide for the docs website (`docs/guides`)
- Is this general enough that any external repo could use it, or is it more specific to our k8 cluster?
- I didn't realize there would be so much to adding a helm chart! It seems like the functionality is pretty independent of metacatui. I wonder if we should re-consider making it it's own repo? One advantage is that it would facilitate updating the helm logic without needing to do a MetacatUI release.

File: helm/metacatui-pv.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+kind: PersistentVolume
+metadata:
+  name: metacatui-pv
+spec:
+  accessModes:
+  - ReadWriteOnce
+  capacity:
+    storage: 10Gi
+  hostPath:
+    path: /Users/jones/development/metacatui/src
```
[the remark this answers is not in the document]
Comment (artntek) [reply helm/metacatui-pv.yaml:11]: removed

File: helm/values.yaml (hunk 1 above)
Comment (artntek) [reply helm/values.yaml:39]: yes - changed comment to reflect this

**Comment by artntek** [comment]
> *It seems like the functionality is pretty independent of metacatui. I wonder if we should re-consider making it it's own repo (e.g. metacatui-helm) ? One advantage is that it would facilitate updating the helm logic without needing to do a MetacatUI release.*

Thanks, @robyngit. I'm not opposed to having the helm chart be in its own repo, although the convention for all our other helm-deployed applications (metacat, metadig, indexer, bookkeeper) is to have the helm chart co-located with the code in the same repo.

We were discussing chart releases/versioning with @mbjones yesterday, and concluded they can be decoupled from the software releases/versioning, at least initially until chart development stabilizes. (i.e. we would not need an "official" metacatui release each time we update the chart, and vice-versa). We decided that we can sync the versions & releases at a later date, if we think it's appropriate.

We recognized that having different chart and software version numbering is confusing, but this follows the "industry standard" approach used by bitnami charts, and seems to be the only viable solution when charts are changing at a different rate than the software they are deploying.

File: helm/values.yaml [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+  ##
+  enabled: true
+
+  ## appConfig.root (required) The url root to be appended after the appConfig.baseUrl, below
+  ##
+  root: "/"
+
+  ## appConfig.theme (required) Corresponds to the name of one of the directories in src/js/themes/
+  ##
+  theme: "default"
```
[the remark this answers is not in the document]
Comment (artntek) [reply helm/values.yaml]: Latest version of metacatui helm chart:
* Worked with Robyn and Matt to resolve how to handle themes & config
* Chart now working with included themes, optional custom theme on a PVC
* Always uses root config (config/config.js) to point to theme and optionally override values
* Optionally provide a custom config.js in a configMap, to override the root config (eg sfwmd/cerp override)

**Pull request merged**